options/options.go

```diff
@@ -36,6 +38,8 @@ type BoilerplateOptions struct {
 	DisableHooks            bool
 	DisableShell            bool
 	DisableDependencyPrompt bool
+	Silent                  bool
+	Logger                  *slog.Logger
```

Comment:
This doesn't feel right, I think we should either use a global slog.Default() or pass the logger instance as a function argument everywhere.